File: playwright-csharp/tests/Pages/RestfulBooker/Booking/HomePage.cs
using System.Globalization;
using Microsoft.Playwright;
using TestTracks.Playwright.CSharp.Specs.Data.DataBuilders.RestfulBooker;
using TestTracks.Playwright.CSharp.Configuration;
using TestTracks.Playwright.CSharp.UI.Common;

namespace TestTracks.Playwright.CSharp.Specs.Pages.RestfulBooker.Booking;

public sealed class HomePage : BasePage
{
    public HomePage(IPage page, TestSettings settings)
        : base(page, settings)
    {
    }

    public async Task OpenBookingForRoom(int roomId, BookingDateRange bookingDates)
    {
        var checkIn = bookingDates.CheckIn.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
        var checkOut = bookingDates.CheckOut.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);

        await OpenAsync($"reservation/{roomId}?checkin={checkIn}&checkout={checkOut}");
    }
}

File: playwright-java/src/test/java/testtracks/pages/restfulbooker/booking/HomePage.java
package testtracks.pages.restfulbooker.booking;

import com.microsoft.playwright.Page;
import java.time.format.DateTimeFormatter;
import testtracks.configuration.TestSettings;
import testtracks.data.builders.restfulbooker.BookingDateRange;
import testtracks.ui.common.BasePage;

public final class HomePage extends BasePage {
  private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ISO_LOCAL_DATE;

  public HomePage(Page page, TestSettings settings) {
    super(page, settings);
  }

  public void openBookingForRoom(int roomId, BookingDateRange bookingDates) {
    var checkIn = bookingDates.checkIn().format(DATE_FORMAT);
    var checkOut = bookingDates.checkOut().format(DATE_FORMAT);

    open("reservation/" + roomId + "?checkin=" + checkIn + "&checkout=" + checkOut);
  }
}
